### PR TITLE
Issue504 locale

### PR DESCRIPTION
**Pull request by JordanTheriault**
Added functionality which upon language change in thimble adds the locale code to Backet's query string. Addresses #504.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -62,6 +62,14 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
         }));
       }
 
+      //We add the localization code to the query params through a URL object
+      //and set search prop to nothing forcing query to be used during url.format()
+      var urlObj, thimbleUrl;
```
Comment (humphd) [inline routes/index.js]: Break this onto two separate lines for increased visibility:

```js
var urlObj;
var thimbleUrl;
```

File: routes/index.js [hunk 2]
```diff
@@ -62,6 +62,14 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
...
+      urlObj = url.parse(req.url, true);
+      urlObj.search = "";
+      urlObj.query["locale"] = req.localeInfo.lang;
+      thimbleUrl = url.format(urlObj);
```
Comment (humphd) [inline routes/index.js]: Actually, just declare this here:

```js
var thimbleUrl = url.format(urlObj);
```

File: routes/index.js [hunk 3]
```diff
@@ -62,6 +62,14 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
...
+
       // We forward query string params down to the editor iframe so that
       // it's easy to do things like enableExtensions/disableExtensions
       var queryString = url.parse(req.url).search || '';
```
Comment (humphd) [inline routes/index.js]: Should this not be `thimbleUrl` instead of `req.url`?

Comment (JordanTheriault) [inline routes/index.js]: Yes, good catch. I must have accidentally removed the change when squashing.

File: routes/index.js [hunk 4]
```diff
@@ -62,9 +62,17 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
         }));
       }
 
+      //We add the localization code to the query params through a URL object
+      //and set search prop to nothing forcing query to be used during url.format()
+      var thimbleUrl;
```
Comment (humphd) [inline routes/index.js]: Don't declare this here, do it all at once on line 71

File: routes/index.js [hunk 5]
```diff
@@ -62,9 +62,17 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
...
+      var urlObj = url.parse(req.url, true);
+      urlObj.search = "";
```
Comment (humphd) [inline routes/index.js]: Why do you clear this?  Won't we lose things like `?enableExtensions=...` and the like?

Comment (humphd) [inline routes/index.js]: OK, I understand this now, it's good.

**Comment by humphd** [comment]
One nit, and this is r+.  Get @sedge to sign off on it, too.

**Comment by sedge** [comment]
r+

**Pull request merged**